### PR TITLE
[prometheus][remote_write] Fix remote_write ingest pipeline to include both metric path 'prometheus.*' and 'prometheus.metrics.*' to fingerprint calculation

### PR DESCRIPTION
**Pull request by tetianakravchenko**
## What does this PR do?

Fix remote_write ingest pipeline to include both metric path 'prometheus.*' and 'prometheus.metrics.*' to fingerprint calculation

There are 2 use cases:
1. If `Use Types` is enabled - metrics will be added under `prometheus.*` in this format:
```
{ "prometheus":
  "metric1": {
    "counter": x
  },
  "metric2": {
    "value": y
  }
  "labels": { 
    ...
  }
}
```

2. If `Use Types` is disabled - metrics will be added under `prometheus.metrics.*` in this format:
```
{ "prometheus":
  "metrics": {
    "metric1": x
    "metric2": y
  }
  "labels": { 
    ...
  }
}
```

This fix is needed to support both cases, otherwise when TSDB is enabled, some documents will be dropped in case `Use Types` is disabled

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

-

## Screenshots

**Previous version - 1.12.0**
> For demonstration purpose I've remove `remove` processor in the local setup - https://github.com/elastic/integrations/blob/ce0f9ec649f50ee1dbcdc2a951e0963543f1db38/packages/prometheus/data_stream/remote_write/elasticsearch/ingest_pipeline/default.yml#L29-L31

`Use_type` disabled:
Screenshot 2023-09-25 at 16 26 19

In this case fingerprint based on the `[metric, labels]` object is not unique that can cause some dropping of documents.

`Use_type` enabled:
Screenshot 2023-09-25 at 16 29 38

**Upgrade prometheus version to 1.12.1**

Screenshot 2023-09-25 at 16 31 41
Upgrade caused recreation of the ingest pipeline, I've removed `remove` processor.

`Use_type` enabled:
Screenshot 2023-09-25 at 16 33 51

`Use_type` disabled (notice that metrics are stored under `prometheus.metric`):
Screenshot 2023-09-25 at 16 35 39

### CODE REVIEW

File: packages/prometheus/data_stream/remote_write/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -7,6 +7,15 @@ processors:
       allow_duplicates: false
       description: Create an empty field of the object type to temporarily store list of all metrics names
   - foreach:
+      if: ctx.prometheus?.metrics != null
+      field: "prometheus.metrics"
+      processor:
+        append:
+          field: "prometheus.labels.metrics_names"
```
Comment (ChrsMark) [inline packages/prometheus/data_stream/remote_write/elasticsearch/ingest_pipeline/default.yml:14]: Is this a field that we add? Will the document end up having the `prometheus.labels.metrics_names` as an actual field?

Comment (tetianakravchenko) [reply packages/prometheus/data_stream/remote_write/elasticsearch/ingest_pipeline/default.yml:14]: no, it is deleted later in the ingest pipeline - https://github.com/elastic/integrations/blob/ce0f9ec649f50ee1dbcdc2a951e0963543f1db38/packages/prometheus/data_stream/remote_write/elasticsearch/ingest_pipeline/default.yml#L29-L31

**Review by ChrsMark — APPROVED** [review]
lgtm. Would be nice if we can cover these through "unit" testing, not sure if we have already or if it's doable though.

**Comment by gizas** [comment]
I am missing only some evidence for tests that the fix worked. Some screenshots only would have been ok

**Review by gizas — APPROVED** [review]

**Comment by tetianakravchenko** [comment]
> I am missing only some evidence for tests that the fix worked. Some screenshots only would have been ok

@gizas I've updated the PR description, could you please have a look?

**Pull request merged**